### PR TITLE
Sponsor hacker view

### PR DESCRIPTION
**Pull request by cktang88**
adds filtering for events that the company owns

### CODE REVIEW

File: src/server/resolvers/index.ts [hunk 1]
```diff
@@ -525,10 +525,9 @@ export const resolvers: CustomResolvers<Context> = {
 
 			if (ctx.user.userType === UserType.Sponsor) {
 				checkIsAuthorized(UserType.Sponsor, ctx.user);
-				return getCompanyEvents(
-					(ctx.user as SponsorDbObject).company._id.toHexString(),
-					ctx.models
-				);
+				const { _id } = (ctx.user as SponsorDbObject).company;
+				const events = await ctx.models.Events.find({ 'owner._id': new ObjectID(_id) }).toArray();
+				return events;
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Why replace the helper function?

File: src/server/events/index.ts [hunk 2]
```diff
@@ -124,10 +124,10 @@ export async function addOrUpdateEvent(
 }
 
 export async function getCompanyEvents(
-	companyID: string,
+	companyID: ObjectID,
```
Comment (slowjazz) [inline src/server/events/index.ts]: Why replace type here?

Comment (cktang88) [reply src/server/events/index.ts]: db IDs should always be objectID if possible, doesn't make sense to be string?

Comment (slowjazz) [reply src/server/events/index.ts]: react hooks can't return ObjectIDs in queries, frontend components don't have ObjectID nor should they - this function then can't be used in frontend components?

Comment (cktang88) [reply src/server/events/index.ts]: yeah this is a serverside function though

**Pull request merged**